pkg/cpi/mockingUtils.go

```diff
@@ -433,9 +471,119 @@ func GetFunctionNameByTestTypeAndMethod(method, testType string) string {
+	}
+	if method == "POST" {
+		return "UploadIntegrationDesigntimeArtifact"
+	}
+	return ""
+}
+
+//GetIntegrationArtifactDeployStatusMockResponse -Provide http respose body
+func GetIntegrationArtifactDeployStatusMockResponse(testType string) (*http.Response, error) {
+
+	return GetMockResponseByTestTypeAndMockFunctionName("GetIntegrationArtifactDeployStatusMockResponse", testType)
+}
+
+//GetIntegrationArtifactDeployErrorDetailsMockResponse -Provide http respose body
+func GetIntegrationArtifactDeployErrorDetailsMockResponse(testType string) (*http.Response, error) {
+
+	return GetMockResponseByTestTypeAndMockFunctionName("GetIntegrationArtifactDeployErrorDetailsMockResponse", "PostiveAndGetDeployedIntegrationDesigntimeArtifactErrorResBody")
+}
+
+//GetIntegrationArtifactDeployStatusMockResponseBody -Provide http respose body
+func GetIntegrationArtifactDeployStatusMockResponseBody() (*http.Response, error) {
+
+	resp := http.Response{
+		StatusCode: 200,
+		Body: ioutil.NopCloser(bytes.NewReader([]byte(`{
+			"d": {
+				"__metadata": {
+					"id": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')",
+					"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')",
+					"media_src": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')/$value",
+					"edit_media": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')/$value"
+				},
+				"Id": "flow1",
+				"Version": "1.0.24",
+				"Name": "flow1",
+				"Status": "STARTED",
+				"ErrorInformation": {
+					"__deferred": {
+						"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')/ErrorInformation"
+					}
+				}
+			}
+		}`))),
+	}
+	return &resp, nil
+}
+
+//GetIntegrationArtifactDeployStatusErrorMockResponseBody -Provide http respose body
+func GetIntegrationArtifactDeployStatusErrorMockResponseBody() (*http.Response, error) {
+
+	resp := http.Response{
+		StatusCode: 200,
+		Body: ioutil.NopCloser(bytes.NewReader([]byte(`{
+			"d": {
+				"__metadata": {
+					"id": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')",
+					"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')",
+					"media_src": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')/$value",
+					"edit_media": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')/$value"
+				},
+				"Id": "smtp",
+				"Version": "2.0",
+				"Name": "smtp",
+				"Status": "ERROR",
+				"ErrorInformation": {
+					"__deferred": {
+						"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')/ErrorInformation"
+					}
+				}
+			}
+		}`))),
```

Comment:
This byte string is very similar the one defined in the `GetIntegrationArtifactDeployStatusMockResponseBody`-function. I suggest having a method that gets the differences as a variable and returns this byte string.

Reply:
done